.github/workflows/build.yml

```diff
@@ -16,9 +16,27 @@ on:
     - 3.7
 
 jobs:
+  check_source:
+    name: 'Check for source changes'
+    runs-on: ubuntu-latest
+    outputs:
+      run_tests: ${{ steps.check.outputs.run_tests }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Check for source changes
+        id: check
+        run: |
+          if [ -z "GITHUB_BASE_REF" ]; then
+            echo '::set-output name=run_tests::true'
+          else
+            git fetch origin master
```

Comment:
You should use $GITHUB_BASE_REF here, no? Also, if you don't specify `--depth=1`, it downloads the whole history, no? It seems like fetching a single commit of master is enough to compute `git diff --name-only origin/$GITHUB_BASE_REF...`.

Reply:
Is it possible that missing commits due to the `--depth=1` causes `fatal: origin/master...HEAD: no merge base` later, as in https://github.com/python/core-workflow/issues/373?